code.rb

```diff
@@ -0,0 +1,178 @@
+      end
+    end
+    resp
+  end
+
+  def my_any?(arg = nil)
+    resp = false
+    return false if to_a.nil?
+    return true if self.class.to_s == 'Hash' && !block_given?
+
+    if block_given?
+      to_a.my_each { |val| resp = true if yield(val) }
+    elsif arg.nil?
+      to_a.my_each { |val| resp = true if val }
+    else
+      case arg.class.to_s
+      when 'Regexp'
+        puts 'regex'
+        to_a.my_each { |val| resp = true if arg.match? val.to_s }
+      when 'Class'
+        puts 'class'
+        to_a.my_each { |val| resp = true if val.is_a? arg }
+      else
+        puts 'else'
+        to_a.my_each { |val| return resp = true if val == arg }
+      end
+    end
+    resp
+  end
+
+  def my_none?(arg = nil)
+    resp = true
+    return true if to_a.nil?
+    return false if self.class.to_s == 'Hash' && !block_given?
+
+    if block_given?
+      to_a.my_each { |val| resp = false if yield(val) }
+    elsif arg.nil?
+      to_a.my_each { |val| resp = false if val }
+    else
+      case arg.class.to_s
+      when 'Regexp'
+        to_a.my_each { |val| resp = false if arg.match? val.to_s }
+      when 'Class'
+        to_a.my_each { |val| resp = false if val.is_a? arg }
+      else
+        to_a.my_each { |val| return resp = false if val == arg }
+      end
+    end
+    resp
+  end
+
+  def my_count(arg = nil)
+    arr = self
+    return arr.length unless block_given? || arg
+
+    counter = 0
+    if arg
+      arr.my_each { |val| counter += 1 if val == arg }
+    else
+      arr.my_each { |val| counter += 1 if yield(val) }
+    end
+    counter
+  end
+
+  def my_map(a_proc = nil)
+    arr = self
+    return to_enum(:my_map) unless block_given?
+
+    if a_proc.nil?
+      arr.my_each_with_index { |v, i| arr[i] = yield(v) }
+    else
+      arr.my_each_with_index { |v, i| arr[i] = a_proc.call(v) }
+    end
+    arr
+  end
+
+  def my_inject(cont = nil, symb = nil)
+    if (cont.is_a? Symbol) || (symb.is_a? Symbol)
+      if cont.is_a? Symbol
+        case cont
+        when :+
+          c = 0
+          my_each { |elem| c += elem }
+        when :-
+          c = self[0]
+          self[1..-1].my_each { |elem| c -= elem }
+        when :*
+          c = self[0]
+          self[1..-1].my_each { |elem| c *= elem }
+        when :/
+          c = self[0]
+          self[1..-1].my_each { |elem| c /= elem }
+        end
+        c
+      elsif cont.is_a? Numeric
+        case symb
+        when :+
+          my_each { |elem| cont += elem }
+        when :-
+          my_each { |elem| cont -= elem }
+        when :*
+          my_each { |elem| cont *= elem }
+        when :/
+          my_each { |elem| cont /= elem }
+        end
+        cont
+      else
+        "undefined method for #{cont}:#{cont.class}"
+      end
+    elsif block_given?
+      cont ||= 0
+      my_each { |elem| cont = yield(cont, elem) }
+      cont
+    else
+      'no block given (LocalJumpError)'
+    end
+  end
+end
```

Comment:
* [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
* [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
  ```ruby
    range = Range.new(5, 50)
    p range.my_inject(4) { |prod, n| prod * n } # returns this error
    # NoMethodError:
       undefined method `length' for 5..50:Range
     # ./code.rb:5:in `my_each'
     # ./code.rb:166:in `my_inject'
  ```
* [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```ruby
    range = Range.new(5, 50)
    p range.my_inject(2, :*) # returns this error
    # NoMethodError:
       undefined method `length' for 5..50:Range
     # ./code.rb:5:in `my_each'
     # ./code.rb:156:in `my_inject'
  ```